src/message.rs

```diff
@@ -135,69 +128,49 @@ impl Value {
     }
 }
 
-/// A transaction definition.
+/// A message definition.
 #[derive(Debug, Getters, Setters, Clone)]
 #[getset(get = "pub", set = "pub(crate)")]
-pub struct Transaction {
-    /// The transaction hash.
-    pub(crate) hash: Hash,
-    /// The transaction address.
-    pub(crate) address: Address,
-    /// The transaction amount.
-    pub(crate) value: Value,
-    /// The transaction tag.
-    pub(crate) tag: Tag,
+pub struct Message {
+    /// The message version.
+    pub(crate) version: u64,
+    /// Message id of the first message this message refers to.
+    pub(crate) trunk: Hash,
+    /// Message id of the second message this message refers to.
+    pub(crate) branch: Hash,
+    /// Length of the payload.
+    pub(crate) payload_length: u64,
+    /// Transaction amount.
+    pub(crate) payload: Payload,
     /// The transaction timestamp.
     pub(crate) timestamp: DateTime<Utc>,
-    /// The transaction current index in the bundle.
-    pub(crate) current_index: u64,
-    /// The transaction last index in the bundle.
-    pub(crate) last_index: u64,
-    /// The transaction bundle hash.
-    pub(crate) bundle_hash: Hash,
-    /// The trunk transaction hash.
-    pub(crate) trunk_transaction: Hash,
-    /// The branch transaction hash.
-    pub(crate) branch_transaction: Hash,
-    /// The transaction nonce.
-    pub(crate) nonce: String,
+    /// Transaction nonce.
+    pub(crate) nonce: u64,
     /// Whether the transaction is confirmed or not.
     pub(crate) confirmed: bool,
     /// Whether the transaction is broadcasted or not.
     pub(crate) broadcasted: bool,
 }
 
-impl Transaction {
-    pub(crate) fn from_bundled(hash: Hash, tx: BundledTransaction) -> crate::Result<Self> {
-        let transaction = Self {
-            hash,
-            address: AddressBuilder::new()
-                .address(tx.address().clone())
-                .key_index(0)
-                .balance(0)
-                .build()?,
-            value: Value {
-                value: *tx.value().to_inner(),
-                unit: ValueUnit::I,
-            },
-            tag: Tag {
-                tag: tx.tag().clone(),
-            },
-            timestamp: DateTime::<Utc>::from_utc(
-                NaiveDateTime::from_timestamp(*tx.attachment_ts().to_inner() as i64, 0),
-                Utc,
-            ),
-            current_index: *tx.index().to_inner() as u64,
-            last_index: *tx.last_index().to_inner() as u64,
-            trunk_transaction: *tx.trunk(),
-            branch_transaction: *tx.branch(),
-            bundle_hash: *tx.bundle(),
-            nonce: "TX NONCE".to_string(), // TODO
+impl Message {
+    pub(crate) fn from_iota_message(message: IotaMessage) -> crate::Result<Self> {
+        let message = Self {
+            version: 1,
+            trunk: message.trunk().clone(),
+            branch: message.branch().clone(),
```

Comment:
trunk and branch fields are public iirc. So you could just move them instead of cloning.